transports/p2p/addrmgr/addrmanager.go

```diff
@@ -211,6 +227,42 @@ func (a *AddrManager) updateAddress(netAddr, srcAddr *wire.NetAddress) {
 	a.Log.Trace().Msgf("Added new address %s for a total of %d addresses", addr, a.nTried+a.nNew)
 }
 
+// BanAddress creates a ban for defaultBanTime for a peer that we cannot connect to.
+func (a *AddrManager) BanAddress(addr string) {
+	a.addrBanned[addr] = time.Now().Add(defaultBanTime)
+
+	// remove from addrTried buckets
+	for _, bucket := range a.addrTried {
+		for e := bucket.Front(); e != nil; e = e.Next() {
+			ka := e.Value.(*KnownAddress)
+			if NetAddressKey(ka.NetAddress()) == addr {
+				bucket.Remove(e)
+				ka.refs--
+				if ka.refs == 0 {
+					a.nTried--
+					delete(a.addrIndex, NetAddressKey(ka.NetAddress()))
+				}
+			}
+		}
+	}
+
+	// remove from addrNew buckets
+	for i, bucket := range a.addrNew {
+		for k, v := range bucket {
+			if k == addr {
+				delete(a.addrNew[i], k)
+				v.refs--
+				if v.refs == 0 {
+					a.nNew--
+					delete(a.addrIndex, k)
+				}
+			}
+		}
+	}
+
+	a.Log.Warn().Msgf("Address %s banned!", addr)
```

Comment:
Maybe let's put a date in this log, something like:
(pseudocode)
`a.Log.Warn().Msgf("Address %s banned until %v!", addr, banDuration)`

Reply:
MINOR

Reply:
Done :)